### ISSUE TITLE
IO write should return the number of bytes written

### ISSUE THREAD

**Issue by stakach** [spider-gazelle/bindata#6]
as per the changes in crystal 0.35.0
currently we are returning 0 as a stop gap

**Comment by Blacksmoke16** [comment]
@stakach This needs revisited due to the release of `0.35.1`.  

```cr
Showing last frame. Use --error-trace for full trace.

In lib/bindata/src/bindata/bitfield.cr:162:27

162 | def write(io, format) : Int64
^
Error: method must return Int64 but it is returning Nil
```

`#write` returns `nil` now, making it incompatible with the return type restriction.

**Comment by stakach** [comment]
@Blacksmoke16 thanks for letting me know
released v1.5.5 which is compatible with both 0.35.0 and 0.35.1
